packages/api/amplify_api/example/integration_test/graphql/iam_test.dart

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+import 'dart:convert';
+
+import 'package:amplify_api/amplify_api.dart';
+import 'package:amplify_api_example/models/ModelProvider.dart';
+import 'package:amplify_flutter/amplify_flutter.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+import '../util.dart';
+
+void main({bool useExistingTestUser = false}) {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('GraphQL IAM', () {
```

Comment:
Is there a way to test custom headers in any of these? Maybe with a lambda resolver? I'm not sure what the use cases are but at some point we'll want to make sure we have a test in place for those.

Reply:
I think the easies way to test that would be to keep some test that uses the headers to do custom auth in cognito mode (like the test utils currently do) and keep that in place even when we replace that util to use `authorizationMode`. I will add a test to do that explicitly here.

Reply:
added test like this